wasmtime-jni/src/io_github_kawamuray_wasmtime_Config/imp.rs

```diff
@@ -0,0 +1,198 @@
+use crate::errors;
+use crate::interop;
+use crate::io_github_kawamuray_wasmtime_Config::JniConfig;
+use jni::objects::{JClass, JObject, JString, JValue};
+use jni::sys::{jboolean, jlong, jobject};
+use jni::{self, JNIEnv};
+use std::path::Path;
+use std::result::Result;
+use wasmtime::{Config, OptLevel, ProfilingStrategy, Strategy};
+
+pub(super) struct JniConfigImpl;
+
+impl<'a> JniConfig<'a> for JniConfigImpl {
+    type Error = errors::Error;
+    fn cache_config_load(
+        env: &JNIEnv,
+        this: JObject,
+        path: JObject,
+    ) -> Result<jobject, Self::Error> {
+        let mut config = interop::get_inner::<Config>(env, this)?;
+        let re = env.call_method(path, "toString", "(V)Ljava/lang/String;", &[]);
+        let j_obj = re.unwrap().l().unwrap();
+        let j_str = JString::from(j_obj);
+        let path_java_str = env.get_string(j_str).unwrap();
+        let path_str = path_java_str.to_str().unwrap();
+        config.cache_config_load(Path::new(path_str))?;
+        return Ok(this.into_inner());
+    }
+    fn cache_config_load_default(env: &JNIEnv, this: JObject) -> Result<jobject, Self::Error> {
+        let mut config = interop::get_inner::<Config>(env, this)?;
+        config.cache_config_load_default()?;
+        return Ok(this.into_inner());
+    }
+    fn cranelift_debug_verifier(
+        env: &JNIEnv,
+        this: JObject,
+        enable: jboolean,
+    ) -> Result<jobject, Self::Error> {
+        let mut config = interop::get_inner::<Config>(env, this)?;
+        config.cranelift_debug_verifier(enable == 1);
+        return Ok(this.into_inner());
+    }
+    fn cranelift_nan_canonicalization(
+        env: &JNIEnv,
+        this: JObject,
+        enable: jboolean,
+    ) -> Result<jobject, Self::Error> {
+        let mut config = interop::get_inner::<Config>(env, this)?;
+        config.cranelift_nan_canonicalization(enable == 1);
+        return Ok(this.into_inner());
+    }
+    fn cranelift_opt_level(
+        env: &JNIEnv,
+        this: JObject,
+        level: JObject,
+    ) -> Result<jobject, Self::Error> {
+        let mut config = interop::get_inner::<Config>(env, this)?;
+        let ordinal = env.call_method(level, "ordinal", "()I", &[])?.i()?;
+        let optlevel: OptLevel;
+        match ordinal {
+            0 => optlevel = OptLevel::None,
+            1 => optlevel = OptLevel::Speed,
+            2 => optlevel = OptLevel::SpeedAndSize,
+            _ => optlevel = OptLevel::Speed,
+        }
+        config.cranelift_opt_level(optlevel);
+        return Ok(this.into_inner());
+    }
+    fn cranelift_other_flag(
+        env: &JNIEnv,
+        this: JObject,
+        name: JString,
+        value: JString,
+    ) -> Result<jobject, Self::Error> {
+        let mut config = interop::get_inner::<Config>(env, this)?;
+        let name_j_str = env.get_string(name).unwrap();
```

Comment:
here too. please use `?` whereever possible. for exceptional cases that we can't convert errors, please use `.expect("error description")` instead of unwrap for better troubleshooting capability :)